README-CMAKE.md

```diff
@@ -0,0 +1,94 @@
+* WITHOUT_CSCORE (OFF Default)
+  * Enabling this option will cause cscore to not be built. This will also implicitly disable cameraserver, the hal and wpilib as well, irrespective of their specific options. If this is on, the opencv build requirement is removed.
+* WITHOUT_ALLWPILIB (ON Default)
+  * Disabling this option will build the hal and wpilib during the build. The HAL is the simulator hal, unless the external hal options are used. The cmake build has no capability to build for the RoboRIO.
+* USE_EXTERNAL_HAL (OFF Default)
+  * TODO
+* EXTERNAL_HAL_FILE
+  * TODO
+
+## Build Setup
+
+The WPILib Cmake build does not allow in source builds. Because the `build` directory is used by gradle, we recommend a `buildcmake` directory in the root. This folder is included in the gitignore.
+
+Once you have a build folder, run cmake configuration in that build directory with the following command.
+
+```
+cmake path/to/allwpilib/root
+```
+
+If you want to change any of the options, add `-DOPTIONHERE=VALUE` to the cmake command. This will check for any dependencies. If everything works properly this will succeed. If not, please check out the troubleshooting section for help.
+
+## Building
+
+Once you have cmake setup. run `make` from the directory you configured cmake in. This will build all libraries possible. If you have a multicore system, we recommend running make with multiple jobs. The usual rule of thumb is 1.5x the number of cores you have. To run a multiple job build, run the following command with x being the number of jobs you want.
+
+```
+make -jx
+```
+
+## Installing
+
+After build, the easiest way to use the libraries is to install them. Run the following command to install the libraries. This will install them so that they can be used from external cmake projects.
+
+```
+sudo make install
+```
+
+## Using the installed libraries for C++.
+
+Using the libraries from C++ is the easiest way to use the built libraries.
+
+To do so, create a new folder to contain your project. Add the following code below to a `CMakeLists.txt` file in that directory.
+
+```
+cmake_minimum_required(VERSION 3.5)
+project(vision_app) # Project Name Here
+
+find_package(wpilib REQUIRED)
+
+add_executable(my_vision_app main.cpp) # exectuable name as first parameter
+target_link_libraries(my_vision_app cameraserver ntcore cscore wpiutil)
+```
+
+If you are using them, `wpilibc` and `hal` should be added before the `cameraserver` declaration in the `target_link_libraries` function.
+
+Add a `main.cpp` file to contain your code, and create a build folder. Move into the build folder, and run
+
+```
+cmake /path/to/folder/containing/CMakeLists
+```
+
+After that, run `make`. That will create your executable. Then you should be able to run `./my_vision_app` to run your application.
+
+
+## Using the installed libraries for Java
+TODO
```

Comment:
Maybe add a "common errors" as well?  E.g. JNI not found, Java compiler not found, OpenCV not found, etc?